types/jquery/v2/index.d.ts

```diff
@@ -321,7 +321,7 @@ interface JQueryGenericPromise<T> {
      * @param failFilter An optional function that is called when the Deferred is rejected.
      * @see {@link https://api.jquery.com/deferred.then/#deferred-then-doneFilter-failFilter-progressFilter}
      */
-    then<U>(doneFilter: (value?: T, ...values: any[]) => U|JQueryPromise<U>, failFilter?: (...reasons: any[]) => any, progressFilter?: (...progression: any[]) => any): JQueryPromise<U>;
+    then<U>(doneFilter: (value: T, ...values: any[]) => U|JQueryPromise<U>, failFilter?: (...reasons: any[]) => any, progressFilter?: (...progression: any[]) => any): JQueryPromise<U>;
```

Comment:
What happens if we want to resolve a promise without a value, i.e. `.then(() => ...)`?

Reply:
Then you should create the JQueryDeferred with 'void' as type argument: 
`$.Deferred<void>().resolve() `

Reply:
Got it - that makes sense. FWIW, [Angular uses the void type parameter](https://stackoverflow.com/a/54143110/367611) a lot, apparently, for that use case in RxJS.

Reply:
The optionality of `value` does not affect your ability to define an arrow function that ignores the parameter:

```ts
function f(cb: (x: number) => void) {}
f(() => {}); // not an error
```

It's not strictly necessary to type the deferred as `void` for that to work.

Reply:
@rbuckton I'm sorry, I don't see what change do you want to see on this pull request. Do you mean in `jquery-tests.ts`?

```
 var promise = $.Deferred<void>().promise();
....
  promise = promise.then(function () { });
```

If I don't use void as generic type here, I cannot assign without error to the promise variable:

Type 'JQueryPromise<void>' is not assignable to type 'JQueryPromise<{}>'